python/ray/serve/_private/deployment_state.py

```diff
@@ -1899,22 +1879,17 @@ def update(self) -> Tuple[bool, bool]:
         Returns (deleted, any_replicas_recovering).
         """
         deleted, any_replicas_recovering = False, False
+        upscale = []
+        downscale = None
         try:
             # Add or remove DeploymentReplica instances in self._replicas.
             # This should be the only place we adjust total number of replicas
             # we manage.
 
-            running_replicas_changed = self._scale_deployment_replicas()
-
             # Check the state of existing replicas and transition if necessary.
-            running_replicas_changed |= self._check_and_update_replicas()
-
-            # Check if the model_id has changed.
-            running_replicas_changed |= self._multiplexed_model_ids_updated
+            self._check_and_update_replicas()
```

Comment:
I flip the order between `_check_and_update_replicas` and `_scale_deployment_replicas` since I think we should first refresh the status of existing replicas and then scale based on the latest status. For example, its possible that `_check_and_update_replicas` stopped a replica so that `_scale_deployment_replicas` needs to start a new one.

Reply:
That should be fine; I don't think this ordering is important. The only downside is it may take an extra iteration for deployment to initially go from starting -> running.

Reply:
Yea, that's true if starting/initialization is super fast.